File: src/Python/medium/string/problem1410.py
class Solution:
    def entityParser(self, text: str) -> str:
        replace_obj = {
            "&quot;": '"',
            "&apos;": "'",
            "&gt;": ">",
            "&lt;": "<",
            "&frasl;": "/",
            "&amp;": "&",
        }
        for key in replace_obj:
            text = text.replace(key, replace_obj[key])
        return text


text = "&amp; is an HTML entity but &ambassador; is not."

solution = Solution()
solution.entityParser(text)

File: src/JavaScript/medium/string/problem1410.js
/**
 * @param {string} text
 * @return {string}
 */

var entityParser = function (text) {
  let replace_obj = {
    "&quot;": '"',
    "&apos;": "'",
    "&gt;": ">",
    "&lt;": "<",
    "&frasl;": "/",
    "&amp;": "&",
  };

  for (let key in replace_obj) {
    text = text.replaceAll(key, replace_obj[key]);
  }
  return text;
};

// let text = "&amp; is an HTML entity but &ambassador; is not."
// let text = "and I quote: &quot;...&quot;";
// let text = "&amp;amp;"
let text = "&amp;gt;";

entityParser(text);
